.gitignore

```diff
@@ -0,0 +1,10 @@
+# Compiled Source #
+##################
+ftsh
+*.o
+
+# OS generated files #
```

Comment:
Let's not include `.DS_Store` and `*.dSYM` files as they're macOS-specific and we are only supporting the CS dept. linux environments.